content/en/docs/tutorials/configuration/updating-configuration-via-a-configmap.md

```diff
@@ -0,0 +1,322 @@
+{{< /note >}}
+
+## Update environment variables of a Pod via a ConfigMap
+Use the `kubectl create configmap` command to create a ConfigMap from [literal values](/docs/tasks/configure-pod-container/configure-pod-configmap/#create-configmaps-from-literal-values):
+
+```shell
+kubectl create configmap car --from-literal=car=mercedes
+```
+
+Edit the deployment `configmap-test` to add the content of the ConfigMap as an environment variable.
+```shell
+kubectl edit deployment configmap-test
+```
+
+Update the `template.spec` as follows:
+```yaml
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - image: nginx:1.14.2
+        imagePullPolicy: IfNotPresent
+        name: nginx        
+        env:
+          - name: CAR
+            valueFrom:
+              configMapKeyRef:
+                key: car
+                name: car
+        ports:
+        - containerPort: 80
+```
+
+Save the file to apply the change, you should see an output as follows:
+```
+deployment.apps/configmap-test edited
+```
+
+Check the deployment.
+```shell
+kubectl get deployment configmap-test
+```
+
+You should see an output similar to:
+```
+NAME             READY   UP-TO-DATE   AVAILABLE   AGE
+configmap-test   1/1     1            1           12m
+```
+
+Check the pod.
+```shell
+kubectl get pod <pod-name>
+```
+
+You should see an output similar to:
+```
+NAME                              READY   STATUS    RESTARTS       AGE
+configmap-test-5fcb9c99d7-mgx9v   1/1     Running   0              12s
+```
+
+View the value of the environment variable named `CAR`.
+```shell
+kubectl exec configmap-test-5fcb9c99d7-mgx9v -- env | grep CAR
+```
+
+You should see the following output:
+```
+CAR=mercedes
+```
+
+Edit the ConfigMap.
+```shell
+kubectl edit configmap car
+```
+
+In the yaml editor, change the value of key `car` from `mercedes` to `tesla` and save.
+```yaml
+apiVersion: v1
+data:
+  car: tesla
+kind: ConfigMap
+metadata:
+  creationTimestamp: "2023-12-27T09:02:47Z"
+  name: car
+  namespace: default
+  resourceVersion: "1471221"
+  uid: 3ce13cd2-dca1-4d3e-b157-af60b5797dab
+```
+
+You should see the following output:
+```
+configmap/car edited
+```
+
+View the value of the environment variable named `CAR`.
+```shell
+kubectl exec configmap-test-5fcb9c99d7-mgx9v -- env | grep CAR
+```
+
+You should see the following output:
+```
+CAR=mercedes
+```
+
+{{< note >}}
+You will observe that although the value of the key inside the ConfigMap has changed, the environment variable in the Pod still possesses the earlier value. This is because environment variables in a Pod are not updated until the Pod is rolled-over.
+{{< /note >}}
+
+Let's simulate a roll-over for the Pod by adding an annotation to the Deployment's `template.metadata`.
+
+Edit the deployment `configmap-test` to add an annotation to the Pod.
+```shell
+kubectl edit deployment configmap-test
```

Comment:
I'd use `kubectl rollout`.

Reply:
Agreed, this is a much better option. Thanks for pointing it out. I've resolved it.